Addition of cutt.ly shortening service

VarthanV opened the pull request:
I have added the implementation of cutt.ly service. It too gives us the analytics of the link that e shortened such as No of Clicks, No of  Visits from Social Networking Sites etc. I have implemented the "short" method. If the author asks to proceed I will implement the analytics function too. Thanks in Advance !

ellisonleao commented on pyshorteners/base.py:
```diff
@@ -104,7 +104,7 @@ def _post(self, url, data=None, json=None, params=None, headers=None):
         return response
 
-    def short(self, url):
+    def short(self, url ,api_key =None ):
```
`api_key`  is provided upon Shortener object instantion, becoming an object property (`self.api_key`) . Can you remove this kwarg please?

VarthanV replied:
I have removed it as per your advice sir

ellisonleao commented on pyshorteners/shorteners/cuttly.py:
```diff
@@ -0,0 +1,41 @@
+    Params Needed :API key
+
+    
+    They too provide analyitcs for the URL 
```
typo on "analytics"

VarthanV replied:
Changed the spelling

ellisonleao commented on pyshorteners/shorteners/cuttly.py:
```diff
@@ -0,0 +1,41 @@
+    api_url = 'https://cutt.ly/api/api.php '
+
+    def short(self, url, api_key=None):
+            if api_key is None:
```
shoud use `self.api_key` here

ellisonleao commented on pyshorteners/shorteners/cuttly.py:
```diff
@@ -0,0 +1,41 @@
+
+    def short(self, url, api_key=None):
+            if api_key is None:
+                return "Please provide a Valid API key"
```
Shoud raise [`ShorteningErrorException`](https://github.com/ellisonleao/pyshorteners/blob/master/pyshorteners/exceptions.py#L1) here

VarthanV replied:
Raised Shortening Error Exception

VarthanV commented:
Thanks, sir for taking time to review my work. I have made the following changes as per your advice, The changes are
        

- Added Unit Tests.

- Removed the API Key Argument from the "short" method in base.py and started using the args from the Shortener Object Ref: Commit cb1626a.

- Raised Shortening Error Exception at  line num 33

ellisonleao commented on pyshorteners/shorteners/cuttly.py:
```diff
@@ -10,32 +10,31 @@ class Shortener(BaseShortener):
-                return "Please provide a Valid API key"
+    def short(self, url):
+            if self.api_key is None:
+                return BadAPIResponseException("Please provide a valid API Key")
```
should call `raise` instead of `return`

VarthanV commented:
Sir I have made the necessary changes but my pull Request was not merged can u just pls guide me in this
